test/exandra/integration_test.exs

```diff
@@ -328,4 +351,168 @@ defmodule Exandra.IntegrationTest do
     assert {1, _} = TestRepo.update_all(query, [])
     assert TestRepo.reload!(fetched_counter).my_counter == 6
   end
+
+  describe "Embeds" do
+    defmodule UDTWithPK do
+      use Ecto.Schema
+      import Ecto.Changeset
+
+      @primary_key {:id, Ecto.UUID, autogenerate: true}
+      embedded_schema do
+        field :name, :string
+      end
+
+      def changeset(entity, params) do
+        entity
+        |> cast(params, [:name])
+      end
+    end
+
+    defmodule EmbeddedSchema do
+      use Ecto.Schema
+      import Ecto.Changeset
+
+      @primary_key false
+      embedded_schema do
+        field :online, :boolean
+        field :dark_mode, :boolean
+      end
+
+      def changeset(entity, params) do
+        entity
+        |> cast(params, [:online, :dark_mode])
+        |> validate_required([:online, :dark_mode])
+      end
+    end
+
+    defmodule MyEmbeddedSchema do
+      use Ecto.Schema
+      import Ecto.Changeset
+      import Exandra, only: [embedded_type: 2, embedded_type: 3]
+
+      @primary_key false
+      schema "my_embedded_schema" do
+        field :my_name, :string, primary_key: true
+        field :my_bool, :boolean
+        embedded_type(:my_embedded_udt, EmbeddedSchema)
+        embedded_type(:my_embedded_udt_list, EmbeddedSchema, cardinality: :many)
+        embedded_type(:my_pk_udt, UDTWithPK)
+      end
+
+      def changeset(entity, params) do
+        entity
+        |> cast(params, [:my_name, :my_bool, :my_embedded_udt, :my_embedded_udt_list, :my_pk_udt])
+      end
+    end
+
+    test "changeset errors" do
+      assert {:error, %Ecto.Changeset{errors: errors}} =
+               %MyEmbeddedSchema{}
+               |> MyEmbeddedSchema.changeset(%{
+                 "my_name" => "EmBetty",
+                 "my_bool" => true,
+                 "my_embedded_udt" => %{
+                   "dark_mode" => "waffle",
```

Comment:
Just because there is a lot going on here could we add a comment or something noting this is where the value is invalid. Not a big deal but i had to recheck the schema to see why

Reply:
Sure thing!